### PR TITLE
[ADD] Added functionality to export PO

### PR DESCRIPTION
**Pull request by SaumilGauswami**
It'll let user to be able to export purchase order with Article. No. and quantity

### CODE REVIEW

File: web_export_po/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+{
+	'name': 'Export Purchase orders',
+	'category': 'Purchase',
+	'summary': 'To export multiple purchase order',
+	'description': """
+       To export multiple purchase order
+    """,
+	'version': '15.0.1.0.0',
+	'author': 'Ensigncode Software',
```
Comment (euroblaze) [inline web_export_po/__manifest__.py:10]: Always use Simplify-ERP™

File: web_export_po/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	'website': 'https://www.ensigncode.com',
```
Comment (euroblaze) [inline web_export_po/__manifest__.py:11]: Always use https://simplify-erp.de

File: web_export_po/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+		'purchase',
+		'purchase_stock',
+	],
+	'data': [
+		'security/ir.model.access.csv',
+		'views/purchase_order_views.xml',
+		'views/product_supplierinfo_views.xml',
+	],
+	'installable': True,
+	'application': True,
```
Comment (euroblaze) [inline web_export_po/__manifest__.py:23]: Not an application.

File: web_export_po/models/product_supplierinfo.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from odoo import fields, models, api, _
+
+
+class ProductSupplierinfo(models.Model):
+	_inherit = "product.supplierinfo"
+
+	article_no = fields.Char('Article No.',translate=True)
```
Comment (euroblaze) [inline web_export_po/models/product_supplierinfo.py:8]: Add in readme.md where the translations can be effected, under a section "# Translations".

File: web_export_po/models/purchase_order.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	def generate_export_report(self, po_ids):
+		"""
+		Generate report in excel file with all the PO selected
+		:param po_ids: Selected Purchase Orders
+		:return:
+		"""
+		products = list(set(po_ids.mapped('order_line').filtered(lambda x: not x.article_no or not x.product_id.seller_ids).mapped('product_id.display_name')))
+		if products:
+			products = '\n'.join(product for product in products)
+			raise UserError(_("Following products do not have this supplier set up with an article number.\n%s ", (products)))
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:21]: Translatable?

File: web_export_po/models/purchase_order.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		worksheet.write(0, 1, u'Quantity', header_style)
+		worksheet.col(1).width = int(20 * 200)
+		row = 1
+		for line in po_ids.order_line:
+			worksheet.write(row, 0, line.article_no)
+			worksheet.write(row, 1, line.product_uom_qty)
+			row += 1
+		workbook.save(file_data)
+		file_data.seek(0)
+		report_data = base64.b64encode(file_data.read())
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:38]: CSV not Excel.
Most of this code can be saved.

File: web_export_po/models/purchase_order.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+		return report_data
+
+	def action_export_orders(self):
+		"""
+		Let you export the orders
+		:return: Downloaded report
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:44]: Translatable?

File: web_export_po/models/purchase_order.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+		"""
+		po_ids = self.env['purchase.order'].search([('id', 'in', self._context.get('active_ids', []))])
+		if po_ids and len(po_ids) > 10:
+			raise UserError(_('Warning: Limit Exceeded\nPerform operation on 10 or fewer POs at a time to prevent performance issues.'))
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:48]: Make it translatable.

File: web_export_po/models/purchase_order.py [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+		report_data = self.generate_export_report(po_ids)
+		report_id = self.env['po.report'].create({'quick_po_report': report_data})
+		return {
+			"type": "ir.actions.act_url",
+			"target": "self",
+			"url": "/web/content?model=po.report&download=true&field=quick_po_report&filename=exported_po_{}.xls&id={}".format(report_id.id, report_id.id),
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:54]: CSV

File: web_export_po/models/purchase_order.py [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class PurchaseOrderLine(models.Model):
+	_inherit = "purchase.order.line"
+
+	article_no = fields.Char('Article No.', translate=True, compute="_compute_article")
+
+	@api.depends('product_qty', 'product_uom', 'product_id', 'order_id.partner_id')
+	def _compute_article(self):
+		for obj in self:
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:65]: Use more descriptive name for object than "obj".

File: web_export_po/models/purchase_order.py [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+			article_no = ''
+			if obj.product_id and obj.product_id.seller_ids and obj.order_id.partner_id:
+				seller_id = obj.product_id.seller_ids.filtered(lambda x: x.product_id.id == obj.product_id.id and x.name.id == obj.order_id.partner_id.id)
```
Comment (euroblaze) [inline web_export_po/models/purchase_order.py:68]: Use more descriptive name than "x".

File: web_export_po/views/purchase_order_views.xml [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+        <field name="arch" type="xml">
+             <xpath expr="//field[@name='order_line']/tree/field[@name='name']" position="after">
+                <field name="article_no" optional="hide"/>
+            </xpath> 
+        </field>
+    </record>
+
+    <!-- action in purchase order form view  to export purchase order-->
+    <record id="action_export_purchase_order" model="ir.actions.server">
+        <field name="name">Export Quick PurchaseOrder (CSV)</field>
```
Comment (euroblaze) [inline web_export_po/views/purchase_order_views.xml:16]: Add in readme.md where to translate unter "# Translations".

File: web_export_po/models/purchase_order.py [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+import base64
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+	_inherit = "purchase.order"
+
+	def generate_export_report(self, po_ids):
+		"""
+		Generate report in excel file with all the PO selected
```
[the remark this answers is not in the document]
Comment (SaumilGauswami) [reply web_export_po/models/purchase_order.py]: @euroblaze Yes all the string which is used in this module is translatable, please go to settings and export the translation for this module to add translation there.

File: web_export_po/models/purchase_order.py (hunk 8 above)
Comment (SaumilGauswami) [reply web_export_po/models/purchase_order.py:37]: @euroblaze It's translatable, ("_" meaning you can translate it.)

File: web_export_po/models/purchase_order.py (hunk 5 above)
Comment (SaumilGauswami) [reply web_export_po/models/purchase_order.py:21]: @euroblaze Yes

File: web_export_po/models/purchase_order.py (hunk 7 above)
Comment (SaumilGauswami) [reply web_export_po/models/purchase_order.py:33]: @euroblaze Yes

**Pull request merged**